ocpmodels/preprocessing/atoms_to_graphs.py

```diff
@@ -57,101 +52,44 @@ def __init__(
+        return _c_index, _n_index, n_distance, _offsets
 
-    def _pad_arrays(
-        self, atoms, split_n_index, split_n_distances, split_offsets
-    ):
-        """Pads arrays to standardize the length"""
-        # c_index is the center index, the atom to find the neighbors of
-        c_index = np.arange(0, len(atoms), 1)
-        # pad c_index
-        pad_c_index = np.repeat(c_index, self.max_neigh)
-        # add dummy variables to desired array length
-        pad_n_index = np.full((len(atoms), self.max_neigh), self.dummy_index)
-        pad_distances = np.full(
-            (len(atoms), self.max_neigh), float(self.dummy_distance)
-        )
-        pad_offsets = np.zeros((len(atoms), self.max_neigh, 3), dtype=np.int)
+    def _reshape_features(self, c_index, n_index, n_distance, offsets):
+        """Stack center and neighbor index and reshapes distances,
+        takes in np.arrays and returns torch tensors"""
+        edge_index = torch.LongTensor(np.vstack((c_index, n_index)))
+        edge_distances = torch.FloatTensor(n_distance)
+        cell_offsets = torch.LongTensor(offsets)
 
-        # loop over the stucture and replace dummy variables where values exist
-        for i, (n_index, distances, offsets) in enumerate(
-            zip(split_n_index, split_n_distances, split_offsets)
-        ):
-            if len(distances) == self.max_neigh:
-                pad_n_index[i] = n_index
-                pad_distances[i] = distances
-                pad_offsets[i] = offsets
-                continue
-            # padding arrays
-            elif len(distances) < self.max_neigh:
-                n_len = len(distances)
-                # potentially add if n_len == 0: print (increase radius)
-                pad_n_index[i][:n_len] = n_index
-                pad_distances[i][:n_len] = distances
-                pad_offsets[i][:n_len] = offsets
-                continue
-            # removing extra values so the length is equal to max_neigh
-            # values are sorted by distance so only nearest neighbors are kept
-            elif len(distances) > self.max_neigh:
-                # this sorts the list min -> max and returns the indices
-                sorted_dist_i = np.argsort(distances)
-                pad_n_index[i] = n_index[sorted_dist_i[: self.max_neigh]]
-                pad_distances[i] = distances[sorted_dist_i[: self.max_neigh]]
-                pad_offsets[i] = offsets[sorted_dist_i[: self.max_neigh]]
-        return pad_c_index, pad_n_index, pad_distances, pad_offsets
+        # remove distances smaller than a tolerance ~ 0. The small tolerance is
+        # needed to correct for pymatgen's neighbor_list returning self atoms
+        # in a few edge cases.
+        nonzero = torch.nonzero(edge_distances >= 1e-8).flatten()
+        edge_index = edge_index[:, nonzero]
+        edge_distances = edge_distances[nonzero]
+        cell_offsets = cell_offsets[nonzero]
```

Comment:
@abhshkdz regarding your comments of removing 0 distances. I do it as part of the preprocessing here. But it doesn't hurt to leave it in the pbc code for redundancy.

Reply:
Cool! Missed this, sorry. We can skip it from the pbc code then!